zappa/core.py

```diff
@@ -1258,6 +1278,12 @@ def rollback_lambda_function_version(self, function_name, versions_back=1, publi
         """
         response = self.lambda_client.list_versions_by_function(FunctionName=function_name)
 
+        # https://github.com/Miserlou/Zappa/pull/2192
+        if response['Versions'][-1]["PackageType"] == "Image":
+            raise NotImplementedError(
+                "Zappa's rollback functionality is not available for Docker based deployments"
+            )
+
```

Comment:
Here's the response from `lambda_client.get_function(FunctionName='...')`:


```python
{   'Code': {   'ImageUri': 'XXXXX.dkr.ecr.us-east-1.amazonaws.com/lambda-docker-flask:default',
                'RepositoryType': 'ECR',
                'ResolvedImageUri': 'XXXXX.dkr.ecr.us-east-1.amazonaws.com/lambda-docker-flask@sha256:0c135dff39b1c28eb973caf1190f1d6134eda19dad88e15bb3e77189ebbe2630'},
    'Configuration': {   'CodeSha256': '0c135dff39b1c28eb973caf1190f1d6134eda19dad88e15bb3e77189ebbe2630',
                         'CodeSize': 0,
                         'Description': 'Zappa Deployment',
                         'Environment': {'Variables': {}},
                         'FunctionArn': 'arn:aws:lambda:us-east-1:XXXXX:function:test-lambda-docker-flask:23',
                         'FunctionName': 'test-lambda-docker-flask',
                         'LastModified': '2020-12-31T07:05:45.014+0000',
                         'LastUpdateStatus': 'Successful',
                         'Layers': [],
                         'MemorySize': 512,
                         'PackageType': 'Image',
                         'RevisionId': 'da514541-052b-4a78-8cdf-cf9d339cdd30',
                         'Role': 'arn:aws:iam::XXXXX:role/test-lambda-docker-flask-ZappaLambdaExecutionRole',
                         'State': 'Active',
                         'Timeout': 30,
                         'TracingConfig': {'Mode': 'PassThrough'},
                         'Version': '23'},
    'ResponseMetadata': {   'HTTPHeaders': {   'connection': 'keep-alive',
                                               'content-length': '1316',
                                               'content-type': 'application/json',
                                               'date': 'Thu, 31 Dec 2020 '
                                                       '17:58:37 GMT',
                                               'x-amzn-requestid': 'b734d3bb-15de-42ea-b194-d8231cb7aa82'},
                            'HTTPStatusCode': 200,
                            'RequestId': 'b734d3bb-15de-42ea-b194-d8231cb7aa82',
                            'RetryAttempts': 0}}
```

So using this, need to see if we can pass the `ResolvedImageUri` (which points to specific image SHA), and pass that to `lambda_client.update_function_code`